### PR TITLE
feat(ava/data): implement lowess and pcorrtest methods

### PR DESCRIPTION
**Pull request by LAI-X**
### PR includes

- [ ] fixed #0
- [x] add / modify test cases
- [ ] documents, demos

### CODE REVIEW

File: packages/ava/__tests__/unit/data/statistics/index.test.ts [hunk 1]
```diff
@@ -20,7 +24,54 @@ import {
...
+  lowess,
+  pcorrtest,
+} from '../../../../src/data/statistics/stdlib';
+
+expect.extend({
+  toBeCloseToArray(received: number[] | number[][], argument: number[] | number[][]) {
+    const pass = received
+      .map((value: number | number[], i) => {
+        const compareValue = argument[i];
+        if (isArray(value) && isArray(compareValue)) return Math.abs(value[0] - compareValue[0]) < 1e-3;
```
Comment (pddpd) [inline packages/ava/__tests__/unit/data/statistics/index.test.ts:51]: `isArray` and `isNumber` are also data utils. Are u sure to use lodash

File: packages/ava/src/data/statistics/stdlib/types.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/**
+ * function options of LOWESS method
+ */
+export type LOWESSOptions = {
```
Comment (pddpd) [inline packages/ava/src/data/statistics/stdlib/types.ts:4]: *LowessOptions

File: packages/ava/src/data/statistics/stdlib/types.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   * Nonnegative parameter which may be used to reduce the number of computations.
+   * Perform a local regression at intervals of delta data points.
+   * */
+  delta?: number;
+};
+
+/**
+ * LOWESS output
+ */
+export type LOWESSOutput = {
```
Comment (pddpd) [inline packages/ava/src/data/statistics/stdlib/types.ts:23]: *LowessOutput

File: packages/ava/src/data/statistics/stdlib/constants.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { LOWESSOptions, PCorrTestOptions } from './types';
```
Comment (pddpd) [inline packages/ava/src/data/statistics/stdlib/constants.ts:1]: do not use the name `stdlib`, these methods should be a part of `statistics`

File: packages/ava/__tests__/unit/data/statistics/index.test.ts [hunk 5]
```diff
@@ -1,3 +1,7 @@
+import '@testing-library/jest-dom/extend-expect';
```
Comment (pddpd) [inline packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: What is this

Comment (pddpd) [inline packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: Have you tested whether these methods are consistent with the results of `stdlib` before

File: packages/ava/src/data/statistics/stdlib/types.ts (hunk 3 above)
Comment (LAI-X) [reply packages/ava/src/data/statistics/stdlib/types.ts:23]: done

File: packages/ava/src/data/statistics/stdlib/types.ts (hunk 2 above)
Comment (LAI-X) [reply packages/ava/src/data/statistics/stdlib/types.ts:4]: done

File: packages/ava/__tests__/unit/data/statistics/index.test.ts (hunk 1 above)
Comment (LAI-X) [reply packages/ava/__tests__/unit/data/statistics/index.test.ts:51]: already replaced

File: packages/ava/__tests__/unit/data/statistics/index.test.ts (hunk 5 above)
Comment (LAI-X) [reply packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: without it, I can't use "expect.extend"
image

Comment (LAI-X) [reply packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: The results of the lowess method (regression method) are close but not identical.
Other methods have the same result

File: packages/ava/src/data/statistics/stdlib/constants.ts (hunk 4 above)
Comment (LAI-X) [reply packages/ava/src/data/statistics/stdlib/constants.ts:1]: have change to 'algorithms'

File: packages/ava/__tests__/unit/data/statistics/index.test.ts (hunk 5 above)
Comment (pddpd) [reply packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: Could the insight demos work?

File: packages/ava/src/data/statistics/stdlib/constants.ts (hunk 4 above)
Comment (pddpd) [reply packages/ava/src/data/statistics/stdlib/constants.ts]: 🤔 The algorithm should not be a subdirectory of statistics. In `stdlib`, it is `stdlib/stats/pcorrtest`

File: packages/ava/__tests__/unit/data/statistics/index.test.ts (hunk 5 above)
Comment (LAI-X) [reply packages/ava/__tests__/unit/data/statistics/index.test.ts:1]: yes.
before : 
image

after :
image

File: packages/ava/src/data/statistics/stdlib/constants.ts (hunk 4 above)
Comment (LAI-X) [reply packages/ava/src/data/statistics/stdlib/constants.ts]: have removed to data/statistics

File: packages/ava/src/data/statistics/common.ts [hunk 6]
```diff
@@ -0,0 +1,264 @@
+import { assert } from '../utils';
```
Comment (pddpd) [inline packages/ava/src/data/statistics/common.ts:1]: The file should be `rename` instead of `new`. And, we name it `base` is better than now one.

Comment (LAI-X) [reply packages/ava/src/data/statistics/common.ts]: image

**Review by pddpd — APPROVED** [review]
LGTM

**Pull request merged**